

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis &mdash; QTL and BSA in Sorghum Semi-Dwarfism 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Welcome to QTL and BSA in Sorghum Semi-Dwarfism’s documentation!" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> QTL and BSA in Sorghum Semi-Dwarfism
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#software-prerequisites-in-an-environment-yaml-form">Software Prerequisites in an Environment .YAML Form</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#install-manually-in-rstudio">Install manually in RStudio</a></li>
<li class="toctree-l3"><a class="reference internal" href="#data-analysis">Data Analysis</a></li>
</ul>
</li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">QTL and BSA in Sorghum Semi-Dwarfism</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/QTLSorghum.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="a-pbgl-snakemake-workflow-and-quatitative-trait-locus-on-bulk-segregant-analysis">
<h1>A PBGL Snakemake Workflow and Quatitative Trait Locus on Bulk Segregant Analysis</h1>
<hr class="docutils" />
<section id="software-prerequisites-in-an-environment-yaml-form">
<h2>Software Prerequisites in an Environment .YAML Form</h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>name: QTLseqr

channels:
- default
- bioconda
- conda-forge
- r

dependencies:
- r-modeest &gt;<span class="o">=</span> <span class="m">2</span>.3.2
- r-ggplot2 &gt;<span class="o">=</span> <span class="m">2</span>.2.0
- r-gtools
- r-dplyr
- r-readr
- r-tidyr
- r-Rcpp
- r-locfit
- r-knitr
- r-rmarkdown
- r-kableExtra
- r-devtools
- r-data.table
- r-vcfR
- r-optparse
- r-base &gt;<span class="o">=</span> <span class="m">4</span>.1.0
</pre></div>
</div>
<section id="install-manually-in-rstudio">
<h3>Install manually in RStudio</h3>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="c1">#install the dependencies</span>
install.packages<span class="o">(</span>c<span class="o">(</span><span class="s2">&quot;data.table&quot;</span>,
           <span class="s2">&quot;modeest&quot;</span>,
           <span class="s2">&quot;locfit&quot;</span>,
           <span class="s2">&quot;dplyr&quot;</span>,
           <span class="s2">&quot;tidyr&quot;</span>,
           <span class="s2">&quot;vcfR&quot;</span>,
           <span class="s2">&quot;ggplot2&quot;</span><span class="o">)</span>, <span class="nv">dependencies</span><span class="o">=</span>TRUE<span class="o">)</span>

<span class="c1"># install devtools</span>
install.packages<span class="o">(</span><span class="s2">&quot;devtools&quot;</span>, <span class="nv">dependencies</span><span class="o">=</span>TRUE<span class="o">)</span>

<span class="c1"># use devtools to install QTLseqr from github</span>
library<span class="o">(</span><span class="s2">&quot;devtools&quot;</span><span class="o">)</span>
<span class="c1">#devtools::install_github(&quot;bmansfeld/QTLseqr&quot;)</span>
<span class="c1">#devtools::install_github(&quot;warthmann/QTLseqr&quot;)</span>
devtools::install_github<span class="o">(</span><span class="s2">&quot;pbgl/QTLseqr&quot;</span><span class="o">)</span>



<span class="c1">#load dependencies</span>
library<span class="o">(</span><span class="s2">&quot;devtools&quot;</span><span class="o">)</span>
library<span class="o">(</span><span class="s2">&quot;data.table&quot;</span><span class="o">)</span>
library<span class="o">(</span><span class="s2">&quot;dplyr&quot;</span><span class="o">)</span>
library<span class="o">(</span><span class="s2">&quot;tidyr&quot;</span><span class="o">)</span>
library<span class="o">(</span><span class="s2">&quot;vcfR&quot;</span><span class="o">)</span>
library<span class="o">(</span><span class="s2">&quot;ggplot2&quot;</span><span class="o">)</span>

<span class="c1">#load the package</span>
library<span class="o">(</span><span class="s2">&quot;QTLseqr&quot;</span><span class="o">)</span>
</pre></div>
</div>
</section>
<section id="data-analysis">
<h3>Data Analysis</h3>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>     QTLseqr::importFromVCF<span class="o">(</span><span class="nv">file</span> <span class="o">=</span> <span class="s2">&quot;freebayes_D2.filtered.vcf&quot;</span>,highBulk <span class="o">=</span> <span class="s2">&quot;D2_F2_tt&quot;</span>,lowBulk <span class="o">=</span> <span class="s2">&quot;D2_F2_TT&quot;</span>,
     <span class="nv">chromList</span> <span class="o">=</span> c<span class="o">(</span><span class="s2">&quot;Chr01&quot;</span>,<span class="s2">&quot;Chr02&quot;</span>,<span class="s2">&quot;Chr03&quot;</span>,<span class="s2">&quot;Chr04&quot;</span>,<span class="s2">&quot;Chr05&quot;</span>,<span class="s2">&quot;Chr06&quot;</span>,<span class="s2">&quot;Chr07&quot;</span>,<span class="s2">&quot;Chr08&quot;</span>,<span class="s2">&quot;Chr09&quot;</span>,<span class="s2">&quot;Chr10&quot;</span><span class="o">)</span>,
     ,filter<span class="o">=</span>FALSE, <span class="nv">outfile</span><span class="o">=</span>TRUE<span class="o">)</span>
     <span class="c1">#Set High bulk and Low bulk sample names and parser generated file name</span>
     <span class="c1">#The file name is generated from the QTLParser_1_MH function in line 119</span>

     HighBulk &lt;- <span class="s2">&quot;D2_F2_tt&quot;</span>
     LowBulk &lt;- <span class="s2">&quot;D2_F2_TT&quot;</span>
     file &lt;- <span class="s2">&quot;Hall.csv&quot;</span>

     <span class="c1">#Choose which chromosomes/contigs will be included in the analysis,</span>

     Chroms &lt;- c<span class="o">(</span><span class="s2">&quot;Chr01&quot;</span>,<span class="s2">&quot;Chr02&quot;</span>,<span class="s2">&quot;Chr03&quot;</span>,<span class="s2">&quot;Chr04&quot;</span>,<span class="s2">&quot;Chr05&quot;</span>,<span class="s2">&quot;Chr06&quot;</span>,<span class="s2">&quot;Chr07&quot;</span>,<span class="s2">&quot;Chr08&quot;</span>,<span class="s2">&quot;Chr09&quot;</span>,<span class="s2">&quot;Chr10&quot;</span><span class="o">)</span>

     df &lt;-
QTLseqr::importFromTable<span class="o">(</span>
 <span class="nv">file</span> <span class="o">=</span> file,
 <span class="nv">highBulk</span> <span class="o">=</span> HighBulk,
 <span class="nv">lowBulk</span> <span class="o">=</span> LowBulk,
 <span class="nv">chromList</span> <span class="o">=</span> chromList
 <span class="nv">sep</span> <span class="o">=</span> <span class="s2">&quot;,&quot;</span>
<span class="o">)</span>

     <span class="c1">#Filter SNPs based on some criteria</span>
     df_filt &lt;-
 QTLseqr::filterSNPs<span class="o">(</span>
 <span class="nv">SNPset</span> <span class="o">=</span> df,
 <span class="nv">refAlleleFreq</span> <span class="o">=</span> <span class="m">0</span>.20,
 <span class="nv">minTotalDepth</span> <span class="o">=</span> <span class="m">100</span>,
 <span class="nv">maxTotalDepth</span> <span class="o">=</span> <span class="m">400</span>,
 <span class="nv">minSampleDepth</span> <span class="o">=</span> <span class="m">40</span>,
          <span class="nv">minGQ</span> <span class="o">=</span> <span class="m">99</span>,
 <span class="nv">verbose</span> <span class="o">=</span> TRUE
<span class="o">)</span>

     <span class="c1">#Run G&#39; analysis</span>
     df_filt&lt;-QTLseqr::runGprimeAnalysis<span class="o">(</span>
 <span class="nv">SNPset</span> <span class="o">=</span> df_filt,
 <span class="nv">windowSize</span> <span class="o">=</span> <span class="m">5000000</span>,
 <span class="nv">outlierFilter</span> <span class="o">=</span> <span class="s2">&quot;deltaSNP&quot;</span>,
 <span class="nv">filterThreshold</span> <span class="o">=</span> <span class="m">0</span>.1<span class="o">)</span>

     <span class="c1">#Run QTLseq analysis</span>
     df_filt &lt;- QTLseqr::runQTLseqAnalysis<span class="o">(</span>
 <span class="nv">SNPset</span> <span class="o">=</span> df_filt,
 <span class="nv">windowSize</span> <span class="o">=</span> <span class="m">5000000</span>,
 <span class="nv">popStruc</span> <span class="o">=</span> <span class="s2">&quot;F2&quot;</span>,
 <span class="nv">bulkSize</span> <span class="o">=</span> c<span class="o">(</span><span class="m">45</span>, <span class="m">38</span><span class="o">)</span>,
 <span class="nv">replications</span> <span class="o">=</span> <span class="m">10000</span>,
 <span class="nv">intervals</span> <span class="o">=</span> c<span class="o">(</span><span class="m">95</span>, <span class="m">99</span><span class="o">)</span>
     <span class="o">)</span>

     <span class="c1">#Plot</span>
     QTLseqr::plotQTLStats<span class="o">(</span><span class="nv">SNPset</span> <span class="o">=</span> df_filt, <span class="nv">var</span> <span class="o">=</span> <span class="s2">&quot;Gprime&quot;</span>, <span class="nv">plotThreshold</span> <span class="o">=</span> TRUE, <span class="nv">q</span> <span class="o">=</span> <span class="m">0</span>.01<span class="o">)</span>
     QTLseqr::plotQTLStats<span class="o">(</span><span class="nv">SNPset</span> <span class="o">=</span> df_filt, <span class="nv">var</span> <span class="o">=</span> <span class="s2">&quot;deltaSNP&quot;</span>, <span class="nv">plotIntervals</span>  <span class="o">=</span> TRUE<span class="o">)</span>

     <span class="c1">#export summary CSV</span>
     QTLseqr::getQTLTable<span class="o">(</span><span class="nv">SNPset</span> <span class="o">=</span> df_filt, <span class="nv">alpha</span> <span class="o">=</span> <span class="m">0</span>.01, <span class="nb">export</span> <span class="o">=</span> TRUE, <span class="nv">fileName</span> <span class="o">=</span> <span class="s2">&quot;my_BSA_QTL.csv&quot;</span><span class="o">)</span>
</pre></div>
</div>
</section>
</section>
</section>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="Welcome to QTL and BSA in Sorghum Semi-Dwarfism’s documentation!" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2022, Michael Hall

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>